### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by pjazdzewski1990**
screen shot 2016-09-25 at 8 15 22 pm

### CODE REVIEW

File: src/components/RecipeBox.js [hunk 1]
```diff
@@ -42,23 +46,28 @@ class RecipeBox extends Component {
           onMouseOver={this.hoverHandler.bind(this)}
           onMouseOut={this.hoverHandler.bind(this)}>
 
-				  <img src={this.props.recipeImage}></img>
+				  <img src={recipe.image}></img>
 
           <div className="recipeDesc">
-            <img id={name} className="more-icon" src={MoreIcon} onClick={this.openMenuHandler.bind(this)}></img>
-            <h3 className="recipe-box-text">{recipeName}</h3>
+            <img id={name} data-recipe-id={recipe.id} className="more-icon" src={MoreIcon} onClick={this.openMenuHandler.bind(this)}></img>
```
Comment (pjazdzewski1990) [inline src/components/RecipeBox.js]: Passing information as `data-*`

File: src/index.js [hunk 2]
```diff
@@ -1,8 +1,32 @@
...
+  { id: 8, name: 'Bar3', description: 'Baaar', image: require('./images/tortilla.jpg'), isStarred: false },
+  { id: 9, name: 'Baz3', description: 'Baz', image: require('./images/cheeserolls.jpg'), isStarred: false },
+  { id: 10, name: 'Foo4', description: 'Foooooooo', image: require('./images/wings.jpg'), isStarred: false },
+  { id: 11, name: 'Bar4', description: 'Baaar', image: require('./images/tortilla.jpg'), isStarred: false },
+  { id: 12, name: 'Baz4', description: 'Baz', image: require('./images/cheeserolls.jpg'), isStarred: false }
+];
+const recipeStore = createStore(RecipeRootReducer, {
+  RecipeReducer: recipesInitialData,
+  SearchReducer: {query: ''}
+});
```
Comment (pjazdzewski1990) [inline src/index.js:25]: Initial name - had problem with names

File: src/actions/RecipeActionCreators.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/*
+ * Holds action creators - functions that will be used to create actions which will be later passed to reducers
```
Comment (pjazdzewski1990) [inline src/actions/RecipeActionCreators.js]: File from lower case (if it's not 1-1 with a JS class)

File: src/components/RecipeBox.js [hunk 4]
```diff
@@ -1,38 +1,42 @@
...
+import React, { Component, PropTypes } from 'react';
 import ContextMenu from './ReactContextMenu';
 import MoreIcon from '../images/moreIcon.png';
-import StarIcon from '../images/starIcon.png';
+import NotStarredIcon from '../images/emptyStarIcon.png';
+import StarredIcon from '../images/starIcon.png';
 import PdfIcon from '../images/pdfIcon.png';
 
 class RecipeBox extends Component {
     constructor() {
```
Comment (burczu) [inline src/components/RecipeBox.js:20]:
``` JavaScript
static propTypes = {
  recipe: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    isStarred: PropTypes.bool.isRequired
 }).isRequired,
 onStarClick: React.PropTypes.func.isRequired
}
```

File: src/containers/FilteredRecipesContainer.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Contains business logic for Recipes presentation classes
+ */
+
+import { connect } from 'react-redux'
```
Comment (pjazdzewski1990) [inline src/containers/FilteredRecipesContainer.js]: Use `@connect`

File: src/reducers/RecipeReducer.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      const idx = currentState.findIndex(item => item.id == action.recipeId);
+      if(idx >= 0) {
+        const starredRecipe = currentState[idx];
+        // ... update ...
+        starredRecipe.isStarred = !starredRecipe.isStarred;
+        // ... build new state
+        const newState = currentState.slice();
+        newState[idx] = starredRecipe;
+
+        return newState;
```
Comment (pjazdzewski1990) [inline src/reducers/RecipeReducer.js]: Use immutable js

File: src/components/RecipeBox.js (hunk 1 above)
Comment (pjazdzewski1990) [reply src/components/RecipeBox.js]: Should be done with bind

File: src/containers/FilteredRecipesContainer.js (hunk 5 above)
Comment (pjazdzewski1990) [reply src/containers/FilteredRecipesContainer.js]: Turned out to be harder than expected https://github.com/pjazdzewski1990/Learn-JS/commit/8711d187e725091c01e9a185c6e998ee976cd157

**Pull request merged**